src/components/text-input/_text-input.scss

```diff
@@ -90,3 +90,82 @@
     }
   }
 }
+
+@mixin text-input--experimental {
+  .#{$prefix}--text-input {
+    @include reset;
+    @include typescale('zeta');
+    background-color: $field-01;
+    width: 100%;
+    height: rem(40px);
+    padding: 0 $spacing-xl 0 $spacing-md;
+    color: $text-02;
+    border: none;
+  }
+
+  .#{$prefix}--text-input ~ .#{$prefix}--label {
+    @include typescale('omega');
+    padding: 0;
+    font-weight: 400;
+    display: block;
+    padding: $spacing-sm 0 $spacing-3xl 0;
+    margin: 0;
+  }
+
+  .#{$prefix}--text-input:required ~ .#{$prefix}--label::after {
+    content: ' *';
+  }
+
+  .#{$prefix}--text-input::-webkit-input-placeholder {
+    color: $text-03;
+  }
+
+  .#{$prefix}--text-input:hover {
+    color: $text-01;
+    outline: 1px solid $brand-01;
+    outline-offset: -1px;
+  }
+
+  .#{$prefix}--text-input:focus,
+  .#{$prefix}--text-input:active {
+    outline: 2px solid $brand-01;
+    outline-offset: -2px;
+  }
+
+  .#{$prefix}--text-input:disabled {
+    cursor: not-allowed;
+    pointer-events: none;
+    color: $text-03;
+  }
+
+  .#{$prefix}--text-input:disabled ~ .#{$prefix}--label {
+    pointer-events: none;
+  }
+
+  .#{$prefix}--text-input[data-invalid],
+  .#{$prefix}--text-input:invalid {
```

Comment:
What invalid state should look like according to duo.
<img width="304" alt="image" src="https://user-images.githubusercontent.com/11670886/45702097-d0ddf480-bb68-11e8-8bfe-2481e5bf7bbb.png">
- no `*`
- font-weight: regular/400
- font-color: `support-01`
- add error icon to the right inside the field

Reply:
Updated everything except adding the icon. Want to check with other devs on how we are handling html updates for these components. cc: @IBM/carbon-developers

Reply:
Also, is that the correct icon? This is what is in the icon package for error filled. 
![screen shot 2018-09-18 at 1 56 06 pm](https://user-images.githubusercontent.com/2753488/45709858-a715d500-bb4a-11e8-906a-fddc3a64470a.png)